org.eclipse.draw2d/src/org/eclipse/draw2d/geometry/Interval.java

```diff
@@ -0,0 +1,293 @@
+/*******************************************************************************
+ * Copyright (c) 2011, 2023 Google, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Google, Inc. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.draw2d.geometry;
+
+import java.io.Serializable;
+
+/**
+ * Represents interval in 1-dimensional space.
+ *
+ * @author mitin_aa
+ * @author lobas_av
+ * @author scheglov_ke
+ * @coverage gef.draw2d
+ */
+public class Interval implements Serializable {
```

Comment:
to make our live easier for future enhancements of this class should it be final?

Reply:
That's a good point. I don't see the use case of extending geometric concepts like intervals.

I think you mentioned updating to Java 17 for the M2. So perhaps it would make sense to transform this class into a record instead? Intervals would then not only be final, but also immutable.

What do you think?

Reply:
I think for value classes like this one it totally makes sense. The grow and start method would then return new instances or?

Reply:
> The grow and start method would then return new instances

Exactly.

Reply:
I would be interested to get more experience with records. So if it is for you, as many user of this new class, the right way I would go for it. Just started my Eclipse IDE to prepare the Java 17 patch. Hope to have something for you by today.

Reply:
If I remember correctly for record classes you also don't need to implement your own hash and equals methods or?

Reply:
> If I remember correctly for record classes you also don't need to implement your own hash and equals methods or?

That's true. It also creates a simple toString() method.

In addition, it also automatically generates getter methods for all fields. At first glance, that should get rid of a lot of boilerplate code inside this class.

Reply:
I've just pushed the changes. I'm not sure how Tycho will respond to this, but at least the IDE currently shows an error, because the compiler compliance is still set to Java 11.

Reply:
Java 17 PR is underway PR #183. I really like the reduced boilerplate code.